src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -373,14 +374,22 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
         goto out_default_range;
     }
 
+    /* GetAddressRange requires context to be set. On DGXA100 it takes 0.03 us
+     * to push and pop the context associated with address (which should be
+     * non-NULL if we are at this point)*/
+    cuCtxPushCurrent(cuda_mem_ctx);
+
     cu_err = cuMemGetAddressRange(&base_address, &alloc_length,
                                   (CUdeviceptr)address);
```

Comment:
Shall we move `cuCtxPopCurrent(&cuda_popped_ctx);` to here? Because we want to pop the pushed context regardless of the success or failure of `cuMemGetAddressRange`

Reply:
Thanks for the suggestion. @brminich made the same suggestion too.